cdap-watchdog-api/src/main/java/co/cask/cdap/api/metrics/MetricsSystemClient.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.metrics;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * An internal client for interacting with the metrics system.
+ */
+public interface MetricsSystemClient {
+
+  /**
+   * Deletes metrics based on the given {@link MetricDeleteQuery}.
+   * Depending on the implementation, this method may return before metrics are actually
+   * deleted.
+   *
+   * @param deleteQuery a {@link MetricDeleteQuery} to specific what metrics to delete
+   * @throws IOException if failed to issue the delete request
+   */
+  void delete(MetricDeleteQuery deleteQuery) throws IOException;
+
+  /**
+   * Queries aggregated metrics with the max time range.
```

Comment:
using what aggregation? Is it SUM?

Reply:
Yes. And apparently the metrics http endpoint only supports SUM. Updated the javadoc.

https://github.com/cdapio/cdap/blob/develop/cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsQueryHelper.java#L379